notification-service/src/test/java/com/redhat/parodos/notification/service/NotificationRecordServiceTests.java

```diff
@@ -0,0 +1,232 @@
+				null, null);
+		assertEquals(expectedResult, result);
+		Mockito.verify(this.notificationRecordRepository, Mockito.times(1))
+				.findByNotificationUserListContaining(notificationUser, pageable);
+
+		Mockito.reset(this.notificationRecordRepository);
+
+		// Notification User exists, state UNREAD and searchTerm empty
+		Mockito.when(this.notificationUserRepository.findByUsername(userName)).thenReturn(validOptional);
+		Mockito.when(this.notificationRecordRepository
+				.findByReadFalseAndNotificationUserListContaining(notificationUser, pageable))
+				.thenReturn(expectedResult);
+		result = this.notificationRecordServiceImpl.getNotificationRecords(pageable, userName, State.UNREAD, null);
+		assertEquals(expectedResult, result);
+		Mockito.verify(this.notificationRecordRepository, Mockito.times(1))
+				.findByReadFalseAndNotificationUserListContaining(notificationUser, pageable);
+
+		Mockito.reset(this.notificationRecordRepository);
+
+		// Notification User exists, state ARCHIVED and searchTerm empty
+		Mockito.when(this.notificationUserRepository.findByUsername(userName)).thenReturn(validOptional);
+		Mockito.when(this.notificationRecordRepository.findByFolderAndNotificationUserListContaining("archive",
+				notificationUser, pageable)).thenReturn(expectedResult);
+		result = this.notificationRecordServiceImpl.getNotificationRecords(pageable, userName, State.ARCHIVED, "");
+		assertEquals(expectedResult, result);
+		Mockito.verify(this.notificationRecordRepository, Mockito.times(1))
+				.findByFolderAndNotificationUserListContaining("archive", notificationUser, pageable);
+
+		Mockito.reset(this.notificationRecordRepository);
+
+		// Notification User exists, state empty and searchTerm valid
+		Mockito.when(this.notificationUserRepository.findByUsername(userName)).thenReturn(validOptional);
+		Mockito.when(this.notificationRecordRepository.search(notificationUser, searchTerm.toLowerCase(), pageable))
+				.thenReturn(expectedResult);
+		result = this.notificationRecordServiceImpl.getNotificationRecords(pageable, userName, null, searchTerm);
+		assertEquals(expectedResult, result);
+		Mockito.verify(this.notificationRecordRepository, Mockito.times(1)).search(notificationUser,
+				searchTerm.toLowerCase(), pageable);
+
+		Mockito.reset(this.notificationRecordRepository);
+
+		// Notification User exists, state and searchTerm valid - exception expected
+		Mockito.when(this.notificationUserRepository.findByUsername(userName)).thenReturn(validOptional);
+		exception = assertThrows(RuntimeException.class, () -> {
+			this.notificationRecordServiceImpl.getNotificationRecords(pageable, userName, State.UNREAD, searchTerm);
+		});
+		assertEquals("Search by state and search term combined not supported", exception.getMessage());
+	}
+
+	@Test
+	void countNotificationRecords() {
+		String userName = "test-user";
+		List<State> states = Arrays.asList(State.values());
```

Comment:
`Arrays.asList` is unnecessary here

Reply:
Removed this line and used the `State.values()` in the foreach loop